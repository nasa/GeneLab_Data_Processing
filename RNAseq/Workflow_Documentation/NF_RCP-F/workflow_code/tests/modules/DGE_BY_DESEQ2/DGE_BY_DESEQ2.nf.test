nextflow_process {

    name "Test Process DGE_BY_DESEQ2"
    script "modules/DGE_BY_DESEQ2/main.nf"
    process "DGE_BY_DESEQ2"
    
    test("GLDS-194") {
        tag 'DGE_BY_DESEQ2'

        when {
            params {
                // define parameters here. Example:
                use_dummy_gene_counts = true
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets/testdata/GLDS-194/Metadata/GLDS-194_bulkRNASeq_v1_runsheet.csv")
                input[1] = file("test-datasets/testdata/GLDS-194/03-RSEM_Counts/*.genes.results")
                input[2] = [ primary_keytype:'ENSEMBL', has_ercc:true ]
                input[3] = file("https://figshare.com/ndownloader/files/36597114")
                input[4] = file("${ baseDir }/bin/dge_annotation_R_scripts.zip")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.summary,
                process.out.norm_counts,
                process.out.summary_ercc,
                process.out.norm_counts_ercc,
                process.out.version
            ).match()
        }

    }

    test("GLDS-321:55_.ISSUE") {
        tag 'DGE_BY_DESEQ2'

        when {
            params {
                // define parameters here. Example:
                use_dummy_gene_counts = true
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = file("test-datasets/testdata/GLDS-321/Metadata/GLDS-321_bulkRNASeq_v1_runsheet.csv")
                input[1] = file("test-datasets/testdata/GLDS-321/03-RSEM_Counts/*.genes.results")
                input[2] = [ primary_keytype:'TAIR', has_ercc:false ]
                input[3] = file("https://figshare.com/ndownloader/files/36597132")
                input[4] = file("${ baseDir }/bin/dge_annotation_R_scripts.zip")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.summary,
                process.out.norm_counts,
                process.out.version,
            ).match()
        }

    }

}