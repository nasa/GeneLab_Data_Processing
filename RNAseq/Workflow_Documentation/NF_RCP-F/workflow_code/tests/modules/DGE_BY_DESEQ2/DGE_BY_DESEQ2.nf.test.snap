{
    "GLDS-321:55_.ISSUE": {
        "content": [
            [
                "summary.txt:md5,2ae67caf20a32f00b87e0f340a4c505b"
            ],
            [
                [
                    "Normalized_Counts.csv:md5,c148732be1d0b1bb61278bfef612f07b",
                    "RSEM_Unnormalized_Counts.csv:md5,fd101e235076c3ae66c513bc96017b33"
                ]
            ],
            [
                "versions.txt:md5,5fac4f3186014a43b8aa3b41d66b2311"
            ]
        ],
        "timestamp": "2023-07-11T22:30:32+0000"
    },
    "GLDS-194": {
        "content": [
            [
                "summary.txt:md5,6c202fd3c11a747e40a49d1369e8875f"
            ],
            [
                [
                    "Normalized_Counts.csv:md5,b4ba348d5446f8ba546a46b966087c1b",
                    "RSEM_Unnormalized_Counts.csv:md5,931c6070b5e19909929c5a217713500b"
                ]
            ],
            [
                "ERCCnorm_summary.txt:md5,1f77ed6cd1a8435038859b0361f6b047"
            ],
            [
                "ERCC_Normalized_Counts.csv:md5,b1d9d5a546a23b6709a9f8c60548b6a7"
            ],
            [
                "versions.txt:md5,1865b6b8900d83ec7881f58fe301da11"
            ]
        ],
        "timestamp": "2023-07-11T22:30:32+0000"
    }
}